### PR TITLE
doc: add basic java docs and java doc maven plugin

### LINKED ISSUE #19: Add javadoc to all classes
We want to be able to publish the ssi artefact to Maven Central. One of the requirements is to have javadoc strings on all classes.

### PR DESCRIPTION
**Pull request by nitin-vavdiya**
## Description
Basic Java doc added with Java doc maven plugin

It will create a javadoc.jar file with `mvn install` command

To verify Java doc run `./mvnw javadoc:javadoc`

Java dock verification was added with compile goal, which means if there is any issue in Java doc then the code will not compile.

Closes #19

### CODE REVIEW

**Comment by borisrizov-zf** [comment]
Hi @nitin-vavdiya, can you update the contents of the docs to some of the key components? I think we can fix the rest as we go.

**Comment by borisrizov-zf** [comment]
Closes #19

**Review by borisrizov-zf — CHANGES_REQUESTED** [review]

File: src/main/java/org/eclipse/tractusx/ssi/examples/ResolveDIDDoc.java [hunk 1]
```diff
@@ -31,7 +31,16 @@
 import org.eclipse.tractusx.ssi.lib.model.did.DidDocument;
 import org.eclipse.tractusx.ssi.lib.model.did.DidMethod;
 
+/** This is example class to demonstrate did document resolve from given did web url */
 public class ResolveDIDDoc {
+  /**
+   * Resovle document did document.
```
[the remark this answers is not in the document]
Comment (nitin-vavdiya) [reply src/main/java/org/eclipse/tractusx/ssi/examples/ResolveDIDDoc.java]: fixed in new commit

File: src/main/java/org/eclipse/tractusx/ssi/examples/VC.java [hunk 2]
```diff
@@ -39,7 +39,13 @@
 import org.eclipse.tractusx.ssi.lib.proof.LinkedDataProofGenerator;
 import org.eclipse.tractusx.ssi.lib.proof.SignatureType;
 
+/** This is example class to demonstrate how create Verifiable Credentials */
 public class VC {
+  /**
+   * Create vc without proof verifiable credential.
```
[the remark this answers is not in the document]
Comment (nitin-vavdiya) [reply src/main/java/org/eclipse/tractusx/ssi/examples/VC.java]: fixed in new commit

**Comment by nitin-vavdiya** [comment]
suggestions are fixed in new commit

File: src/main/java/org/eclipse/tractusx/ssi/lib/SsiLibrary.java [hunk 3]
```diff
@@ -25,7 +25,9 @@
 import net.i2p.crypto.eddsa.EdDSASecurityProvider;
 import org.bouncycastle.jce.provider.BouncyCastleProvider;
 
+/** The type Ssi library. */
```
Comment (florianrusch-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/SsiLibrary.java:28]:
```suggestion
/** The type SSI library. */
```

**Review by florianrusch-zf — CHANGES_REQUESTED** [review]
I have not yet completed the review.

**Comment by nitin-vavdiya** [comment]
> I have not yet completed the review.

Okay, let me know. Will fix all  suggestions in one go

**Comment by borisrizov-zf** [comment]
@carslen do we need an IP check? The handbook defines "project content" as any kind of content. Although this is just class documentation, it probably still falls under the >1000 SLOC rule.

**Comment by carslen** [comment]
> @carslen do we need an IP check? The handbook defines "project content" as any kind of content. Although this is just class documentation, it probably still falls under the >1000 SLOC rule.

I did not check the hole PR, but if it is only comments, then I'd approve the PR without starting a IP-Check.

**Comment by borisrizov-zf** [comment]
> > @carslen do we need an IP check? The handbook defines "project content" as any kind of content. Although this is just class documentation, it probably still falls under the >1000 SLOC rule.
> 
> I did not check the hole PR, but if it is only comments, then I'd approve the PR without starting a IP-Check.

Yes, it's only Javadoc on each class. Thanks for the help :)

**Comment by borisrizov-zf** [comment]
@nitin-vavdiya please rebase as we've merged two branches.

**Comment by nitin-vavdiya** [comment]
> @nitin-vavdiya please rebase as we've merged two branches.

done

**Comment by florianrusch-zf** [comment]
@nitin-vavdiya please change the target branch to the develop branch

**Comment by nitin-vavdiya** [comment]
> @nitin-vavdiya please change the target branch to the develop branch

there is no develop branch

**Comment by borisrizov-zf** [comment]
> @nitin-vavdiya please change the target branch to the develop branch

yes this repo directly merges into main.

**Comment by florianrusch-zf** [comment]
> > @nitin-vavdiya please change the target branch to the develop branch
> 
> there is no develop branch

ahhh sorry, mixed the repository. :-/

**Comment by florianrusch-zf** [comment]
But why is then this PR adding the following files? They should be already on the main branch.

image

**Comment by borisrizov-zf** [comment]
@nitin-vavdiya seems your rebase didn't work correctly. can you double check?

File: src/main/java/org/eclipse/tractusx/ssi/examples/VP.java [hunk 4]
```diff
@@ -37,8 +37,19 @@
 import org.eclipse.tractusx.ssi.lib.serialization.jwt.SerializedJwtPresentationFactory;
 import org.eclipse.tractusx.ssi.lib.serialization.jwt.SerializedJwtPresentationFactoryImpl;
 
+/**
+ * his is an example class to demonstrate how to create a Verifiable Presentation in JSON-LD and KWT
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/examples/VP.java:41]:
```suggestion
 * This is an example class to demonstrate how to create a Verifiable Presentation in JSON-LD and KWT
```

File: src/main/java/org/eclipse/tractusx/ssi/examples/ResolveDIDDoc.java [hunk 5]
```diff
@@ -31,7 +31,16 @@
 import org.eclipse.tractusx.ssi.lib.model.did.DidDocument;
 import org.eclipse.tractusx.ssi.lib.model.did.DidMethod;
 
+/** This is example class to demonstrate did document resolve from given did web url */
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/examples/ResolveDIDDoc.java:34]:
```suggestion
/** This is an example class to demonstrate did document resolve from given did web url */
```

File: src/main/java/org/eclipse/tractusx/ssi/lib/crypt/IKeyGenerator.java [hunk 6]
```diff
@@ -23,6 +23,14 @@
 
 import org.eclipse.tractusx.ssi.lib.exception.KeyGenerationException;
 
+/** The interface Key generator. */
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/crypt/IKeyGenerator.java:26]:
```suggestion
/** The interface KeyGenerator. */
```

File: src/main/java/org/eclipse/tractusx/ssi/lib/crypt/jwk/JsonWebKey.java [hunk 7]
```diff
@@ -22,36 +22,62 @@
...
 
 import com.nimbusds.jose.jwk.OctetKeyPair;
-import java.io.IOException;
 import lombok.AccessLevel;
 import lombok.RequiredArgsConstructor;
 import org.eclipse.tractusx.ssi.lib.crypt.IPrivateKey;
 import org.eclipse.tractusx.ssi.lib.crypt.IPublicKey;
 import org.eclipse.tractusx.ssi.lib.crypt.octet.OctetKeyPairFactory;
 
+/** The type Json web key. */
```
Comment (borisrizov-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/crypt/jwk/JsonWebKey.java:31]:
```suggestion
/** The type JsonWebKey. */
```

File: src/main/java/org/eclipse/tractusx/ssi/lib/model/base/Base64.java [hunk 8]
```diff
@@ -27,19 +27,38 @@
...
 
+/** The type Base 64. */
 @Value
 @EqualsAndHashCode
 public class Base64 implements MultibaseString {
 
+  /**
+   * Can decode boolean.
+   *
+   * @param encoded the encoded
```
[the remark this answers is not in the document]
Comment (borisrizov-zf) [reply src/main/java/org/eclipse/tractusx/ssi/lib/model/base/Base64.java:38]: This is unresolved due to rebase.

**Review by borisrizov-zf — CHANGES_REQUESTED** [review]
Please recheck old comments from Florian, as the rebase has removed some changes.

**Comment by nitin-vavdiya** [comment]
> Please recheck old comments from Florian, as the rebase has removed some changes.

done

**Comment by carslen** [comment]
@florianrusch-zf please review if your requested changes have been applied.

**Comment by borisrizov-zf** [comment]
@florianrusch-zf ping

File: src/main/java/org/eclipse/tractusx/ssi/lib/crypt/x21559/x21559PublicKey.java [hunk 9]
```diff
@@ -35,21 +35,36 @@
...
   public x21559PublicKey(byte[] publicKey) throws InvalidePublicKeyFormat {
     if (this.getKeyLength() != publicKey.length) {
       throw new InvalidePublicKeyFormat(getKeyLength(), publicKey.length);
     }
     this.originalKey = publicKey;
   }
 
-  public x21559PublicKey(String publicKey, boolean PEMformat)
+  /**
+   * Instantiates a new X 21559 public key.
```
Comment (florianrusch-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/crypt/x21559/x21559PublicKey.java:57]:
```suggestion
   * Instantiates a new X21559 public key.
```

File: src/main/java/org/eclipse/tractusx/ssi/lib/did/resolver/DidDocumentResolverRegistry.java [hunk 10]
```diff
@@ -25,12 +25,32 @@
...
+   *
+   * @param did the did
+   * @return the did document resolver
+   * @throws DidDocumentResolverNotRegisteredException the did document resolver not registered
+   *     exception
+   */
   DidDocumentResolver get(DidMethod did) throws DidDocumentResolverNotRegisteredException;
 
+  /**
+   * Register.
```
Comment (florianrusch-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/did/resolver/DidDocumentResolverRegistry.java:44]:
```suggestion
   * Register a new did document resolver
```

File: src/main/java/org/eclipse/tractusx/ssi/lib/did/resolver/DidDocumentResolverRegistry.java [hunk 11]
```diff
@@ -25,12 +25,32 @@
...
+   *
+   * @param resolver the resolver
+   */
   void register(DidDocumentResolver resolver);
 
+  /**
+   * Unregister.
```
Comment (florianrusch-zf) [inline src/main/java/org/eclipse/tractusx/ssi/lib/did/resolver/DidDocumentResolverRegistry.java:51]:
```suggestion
   * Unregister a did document resolver
```

**Review by florianrusch-zf — CHANGES_REQUESTED** [review]
It's way to much to do a proper review on this...I would say, lets merge it and improve the comments, when revisiting the places in later changes. WDYT? :-/

File: src/main/java/org/eclipse/tractusx/ssi/lib/crypt/jwk/JsonWebKey.java [hunk 12]
```diff
@@ -22,36 +22,62 @@
...
   }
 
+  /**
+   * Gets x.
+   *
+   * @return the x
+   */
   public String getX() {
     return this.keyPair.getX().toString();
   }
```
[the remark this answers is not in the document]
Comment (nitin-vavdiya) [reply src/main/java/org/eclipse/tractusx/ssi/lib/crypt/jwk/JsonWebKey.java:83]: Please check: https://openid.net/specs/draft-jones-json-web-key-03.html

refer:  4.2.1.  JWK Key Object Members for Elliptic Curve Keys

**Comment by borisrizov-zf** [comment]
This can be merged now, with the understanding that we'll improve as we go.

**Comment by florianrusch-zf** [comment]
@nitin-vavdiya please apply the requested changes from @borisrizov-zf

If they got applied, we can continue with the merge.

**Comment by nitin-vavdiya** [comment]
> @nitin-vavdiya please apply the requested changes from @borisrizov-zf
> 
> If they got applied, we can continue with the merge.

Yes, changes are applied

**Comment by florianrusch-zf** [comment]
No, the changes are not applied:
- https://github.com/eclipse-tractusx/SSI-agent-lib/pull/16/files/26951b5a0fe67660f5e988c2f0243e6c6e5a9eea#r1371742216
- https://github.com/eclipse-tractusx/SSI-agent-lib/pull/16/files/26951b5a0fe67660f5e988c2f0243e6c6e5a9eea#r1371742617
- https://github.com/eclipse-tractusx/SSI-agent-lib/pull/16/files/26951b5a0fe67660f5e988c2f0243e6c6e5a9eea#r1371744141
- https://github.com/eclipse-tractusx/SSI-agent-lib/pull/16/files/26951b5a0fe67660f5e988c2f0243e6c6e5a9eea#r1371744665

**Comment by nitin-vavdiya** [comment]
> No, the changes are not applied:
> 
> * https://github.com/eclipse-tractusx/SSI-agent-lib/pull/16/files/26951b5a0fe67660f5e988c2f0243e6c6e5a9eea#r1371742216
> * https://github.com/eclipse-tractusx/SSI-agent-lib/pull/16/files/26951b5a0fe67660f5e988c2f0243e6c6e5a9eea#r1371742617
> * https://github.com/eclipse-tractusx/SSI-agent-lib/pull/16/files/26951b5a0fe67660f5e988c2f0243e6c6e5a9eea#r1371744141
> * https://github.com/eclipse-tractusx/SSI-agent-lib/pull/16/files/26951b5a0fe67660f5e988c2f0243e6c6e5a9eea#r1371744665

done

**Comment by florianrusch-zf** [comment]
build pipeline is running into a timeout while trying to pull a dependency. Will retry tomorrow

**Comment by borisrizov-zf** [comment]
I think the Fraunhofer server is down.

**Review by florianrusch-zf — APPROVED** [review]

**Pull request merged**